src/main/java/it/pagopa/pm/gateway/service/CcPaymentInfoService.java

```diff
@@ -40,46 +50,74 @@ public CcPaymentInfoService(@Value("${vpos.method.notifyUrl}") String methodNoti
         this.clientsConfig = clientsConfig;
     }
 
-    public CcPaymentInfoResponse getPaymentoInfo(String requestId) {
-        Optional<PaymentRequestEntity> paymentInfo = paymentRequestRepository.findByGuidAndRequestEndpoint(requestId, REQUEST_PAYMENTS_VPOS);
+    public CcPaymentInfoResponse getPaymentInfo(String requestId) {
+        Optional<PaymentRequestEntity> paymentInfo = paymentRequestRepository.findByGuidAndRequestEndpoint(requestId, VPOS_AUTHORIZATIONS);
 
         if (!paymentInfo.isPresent()) {
             log.error("No CreditCard request entity has been found for requestId: " + requestId);
             throw new CcHttpException(HttpStatus.NOT_FOUND, "RequestId non trovato");
         }
 
-        return createInfoPaymentResponse(paymentInfo.get());
+        return buildCcPaymentInfoResponse(paymentInfo.get());
     }
 
-    private CcPaymentInfoResponse createInfoPaymentResponse(PaymentRequestEntity paymentInfo) {
+    private CcPaymentInfoResponse buildCcPaymentInfoResponse(PaymentRequestEntity paymentRequestEntity) {
         CcPaymentInfoResponse response = new CcPaymentInfoResponse();
-        String requestId = paymentInfo.getGuid();
-        String idTransaction = paymentInfo.getIdTransaction();
-        response.setStatus(paymentInfo.getStatus());
+        String requestId = paymentRequestEntity.getGuid();
+        PaymentRequestStatusEnum paymentRequestStatusEnum =
+                getEnumValueFromString(paymentRequestEntity.getStatus());
+        response.setPaymentRequestStatusEnum(paymentRequestStatusEnum);
         response.setRequestId(requestId);
-
-        if (AUTHORIZED.name().equals(paymentInfo.getStatus())) {
-            response.setAuthCode(paymentInfo.getAuthorizationCode());
-        } else if(CREATED.name().equals(paymentInfo.getStatus())) {
-            if (METHOD.name().equals(paymentInfo.getResponseType())) {
-                String threeDsMethodData = generate3DsMethodData(requestId);
-                response.setThreeDsMethodData(threeDsMethodData);
-                response.setResponseType(paymentInfo.getResponseType());
-            } else if (CHALLENGE.name().equals(paymentInfo.getResponseType())) {
-                String creq = getCreqFromChallengeUrl(paymentInfo);
+        String authorizationUrl = paymentRequestEntity.getAuthorizationUrl();
+        String[] vposUrlAndCreq = StringUtils.splitByWholeSeparatorPreserveAllTokens(authorizationUrl, CREQ);
+        String vposUrl = null;
+        String creq = null;
+        if (ObjectUtils.isNotEmpty(vposUrlAndCreq)) {
+            vposUrl = vposUrlAndCreq[0];
+            creq = vposUrlAndCreq.length == 2 ? vposUrlAndCreq[1] : null;
+        }
```

Comment:
This snippet manipulates a URL via its string representation and makes the underlying logic a bit less clear (for example here we are assuming that `creq` is optional and can be the only query parameter in the authorization url).
I would suggest to either refactor this to extract the `creq` query param via URL/URI parsing (e.g. see [here](https://stackoverflow.com/questions/13592236/parse-a-uri-string-into-name-value-collection)) or extract this to a separate private method